builtin/providers/aws/resource_aws_internet_gateway.go

```diff
@@ -46,18 +46,40 @@ func resourceAwsInternetGatewayCreate(d *schema.ResourceData, meta interface{})
 	log.Printf("[INFO] InternetGateway ID: %s", d.Id())
 
 	resource.Retry(5*time.Minute, func() *resource.RetryError {
+		log.Printf("[INFO] Refreshing InternetGateway ID: %s", d.Id())
 		igRaw, _, err := IGStateRefreshFunc(conn, d.Id())()
 		if igRaw != nil {
 			return nil
 		}
 		if err == nil {
+			log.Printf("[INFO] RetryableError creating InternetGateway ID: %s %s", d.Id(), err)
 			return resource.RetryableError(err)
 		} else {
+			log.Printf("[INFO] NonRetryableError creating InternetGateway ID: %s %s", d.Id(), err)
 			return resource.NonRetryableError(err)
 		}
 	})
 
-	err = setTags(conn, d)
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+		log.Printf("[INFO] Setting tags InternetGateway ID: %s", d.Id())
+		err = setTags(conn, d)
+		if err == nil {
+			return nil
+		}
+		ec2err, ok := err.(awserr.Error)
+		if !ok {
+			log.Printf("[INFO] RetryableError setting tags InternetGateway ID: %s %s", d.Id(), err)
+			return resource.RetryableError(err)
```

Comment:
If it's _not_ an EC2 Error, it's a `RetryableError` error? It seems like this should be the `if ok` block and just fall through to the `NonRetryableError` below, unless I'm mistaken here